packages/suite/src/views/wallet/sign-verify/components/SignMessage/index.tsx

```diff
@@ -0,0 +1,267 @@
+        address = account.descriptor;
+        path = account.path;
+    }
+
+    const { getValues, setValue, errors, formState, reset, control, register } = useForm<Inputs>({
+        defaultValues: {
+            address,
+            path,
+            signMessage: '',
+            accountsSelect: { label: '', value: '', type: '' },
+            signSignature: '',
+        },
+        mode: 'onChange',
+    });
+    const { isValid } = formState;
+
+    useEffect(() => {
+        return () => {
+            reset(); // clear inputs after coin change
+        };
+    }, [reset]);
+
+    return (
+        <Card title={<Translation id="TR_SIGN_MESSAGE" />} isColumn>
+            <Column>
+                <Row>
+                    {account.networkType === 'bitcoin' && (
+                        <Controller
+                            as={<StyledSelect options={accountsSelectOpt} />}
+                            control={control}
+                            topLabel={<Translation id="TR_ADDRESS" />}
+                            name="accountsSelect"
+                            state={errors.accountsSelect ? 'error' : undefined}
+                            onChange={([selected]) => {
+                                path = selected.value;
+                                address = selected.label;
+                                return selected;
+                            }}
+                            bottomText={
+                                errors.accountsSelect && (
+                                    <Translation
+                                        id={(errors.accountsSelect.type as any) as errorsMessage}
+                                    />
+                                )
+                            }
+                            rules={{
+                                validate: {
+                                    TR_REQUIRED_FIELD: (selectValue: Inputs['accountsSelect']) => {
+                                        return selectValue.value.length > 0;
+                                    },
+                                },
+                            }}
+                        />
+                    )}
+                    {/* input is more convenient from the ability to select and copy */}
+                    {account.networkType === 'ethereum' && (
+                        <Input
+                            topLabel={<Translation id="TR_ADDRESS" />}
+                            name="address"
+                            value={address}
+                            type="text"
+                            innerRef={register}
+                            readOnly
+                        />
+                    )}
+                    <StyledIcon
+                        icon="COPY"
+                        onClick={() => {
+                            // eslint-disable-next-line no-unused-expressions
+                            copyToClipboard(address, null)
```

Comment:
instead of validation address.length inside `copyToClipboard` and adding more logic you should disable this button (do not allow to copy if address is empty or invalid)

Reply:
as I said already, the user does not expect toast messages about successful copying and the button color is gray, the user does not have a reference for comparison this is a disabled button or not, so this decision is confusing in terms of ui, but I rewritten as you wanted